api/handlers/process_test.go

```diff
@@ -454,4 +457,52 @@ var _ = Describe("Process", func() {
 			})
 		})
 	})
+
+	Describe("DELETE /v3/processes/:guid/instances/:instance", func() {
+		var instance string
+		BeforeEach(func() {
+			processRepo.GetProcessReturns(repositories.ProcessRecord{
+				GUID:             "process-guid",
+				AppGUID:          "app-guid",
+				SpaceGUID:        "space-guid",
+				DesiredInstances: 2,
+				Type:             "web",
+			}, nil)
+			processRepo.GetAppRevisionForProcessReturns("0", nil)
+			instance = "0"
+		})
+		JustBeforeEach(func() {
+			req, err := http.NewRequestWithContext(ctx, "DELETE", "/v3/processes/process-guid/instances/"+instance, strings.NewReader("the-json-body"))
+			Expect(err).NotTo(HaveOccurred())
+			routerBuilder.Build().ServeHTTP(rr, req)
+
+		})
+		It("restarts the instance", func() {
+			Expect(rr).To(HaveHTTPStatus(http.StatusNoContent))
+			_, actualAuthInfo, actualAppRevision, actualAppGUID, actualSpaceGUID, actualInstanceID, actualProcessType := podRepo.DeletePodArgsForCall(0)
+			Expect(actualAuthInfo).To(Equal(authInfo))
+			Expect(actualAppRevision).To(Equal("0"))
+			Expect(actualAppGUID).To(Equal("app-guid"))
+			Expect(actualSpaceGUID).To(Equal("space-guid"))
+			Expect(actualInstanceID).To(Equal("0"))
+			Expect(actualProcessType).To(Equal("web"))
+			Expect(podRepo.DeletePodCallCount()).To(Equal(1))
```

Comment:
It is a good practice to assert the call count first and then to check the arguments. The reason is to avoid a confusing "index out of bounds"-like error when invoking `DeletePodArgsForCall` in case the repository is not invoked for some reason.

Reply:
moved the DeletePodCallCount up